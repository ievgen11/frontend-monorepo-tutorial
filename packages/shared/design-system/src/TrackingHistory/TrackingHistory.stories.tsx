import * as React from "react";
import { withKnobs, text } from "@storybook/addon-knobs";

import { TrackingHistory, TrackingHistoryItem } from "./index";

export default {
  title: "TrackingHistory",
  component: TrackingHistory,
  decorators: [withKnobs],
};

export const Default = () => (
  <TrackingHistory>
    <TrackingHistoryItem
      label={text("Date", "Monday")}
      value={text("Value", "Delivered")}
    />
  </TrackingHistory>
);

export const Example = () => (
  <TrackingHistory>
    <TrackingHistoryItem
      label={text("Date1", "Monday")}
      value={text("Value1", "Delivered")}
    />
    <TrackingHistoryItem
      label={text("Date2", "Saturday")}
      value={text("Value2", "Sorted at depot")}
    />
    <TrackingHistoryItem
      label={text("Date3", "Friday")}
      value={text("Value3", "Picked up")}
    />
  </TrackingHistory>
);
