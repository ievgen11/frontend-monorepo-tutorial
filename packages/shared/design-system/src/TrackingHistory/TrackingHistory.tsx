import React from "react";
import styled from "styled-components";

const List = styled.ul`
  list-style: none;
  margin: 24px 0;
  padding: 0;
`;

const Item = styled.li`
  display: flex;
  flex-direction: column;
  position: relative;
  padding: 12px 32px 24px 32px;

  &::after {
    position: absolute;
    content: "";
    left: 12px;
    width: 1px;
    transform: translate(-50%, 0);
    height: 100%;
    background-color: #f0f0f0;
  }

  &::before {
    position: absolute;
    content: "";
    left: 12px;
    top: 12px;
    width: 12px;
    transform: translate(-50%, 0);
    height: 12px;
    border-radius: 12px;
    background-color: #f0f0f0;
  }

  &:last-child {
    padding-bottom: 0;

    &::after {
      display: none;
    }
  }
`;

const Label = styled.span`
  font-size: 13px;
  font-weight: 600;
  margin-bottom: 12px;
`;

const Value = styled.span`
  font-size: 18px;
  font-weight: 300;
`;

export interface TrackingHistoryProps {
  children?: React.ReactNode;
}

export interface TrackingHistoryItemProps {
  label?: React.ReactNode;
  value?: React.ReactNode;
}

export const TrackingHistory: React.FC<TrackingHistoryProps> = ({
  children,
}): JSX.Element => <List>{children}</List>;

export const TrackingHistoryItem: React.FC<TrackingHistoryItemProps> = ({
  label,
  value,
}): JSX.Element => (
  <Item>
    <Label>{label}</Label>
    <Value>{value}</Value>
  </Item>
);
