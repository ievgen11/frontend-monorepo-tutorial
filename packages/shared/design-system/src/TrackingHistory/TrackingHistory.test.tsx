import * as React from "react";
import * as renderer from "react-test-renderer";

import { TrackingHistory, TrackingHistoryItem } from "./TrackingHistory";

describe("<TrackingHistory />", () => {
  test("Snapshot is intact", () => {
    const tree = renderer
      .create(<TrackingHistory>Test</TrackingHistory>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<TrackingHistoryItem />", () => {
  test("Snapshot is intact", () => {
    const tree = renderer
      .create(<TrackingHistoryItem>Test</TrackingHistoryItem>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
