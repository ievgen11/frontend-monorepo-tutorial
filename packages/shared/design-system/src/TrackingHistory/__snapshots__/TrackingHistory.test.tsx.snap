// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackingHistory /> Snapshot is intact 1`] = `
<ul
  className="sc-bdfBwQ jCCheO"
>
  Test
</ul>
`;

exports[`<TrackingHistoryItem /> Snapshot is intact 1`] = `
<li
  className="sc-gsTCUz bmwMqN"
>
  <span
    className="sc-dlfnbm AMCU"
  />
  <span
    className="sc-hKgILt kfnKoE"
  />
</li>
`;
